modules/python/src2/pycompat.hpp

```diff
@@ -174,107 +198,106 @@ PyObject* pyopencv_from(const TYPE& src)
 #endif
 
 
-#define CVPY_TYPE_DECLARE(WNAME, NAME, STORAGE, SNAME) \
-    struct pyopencv_##NAME##_t \
+#define CVPY_TYPE_DECLARE(CLASS_ID, NAME, STORAGE, SNAME, SCOPE) \
+    struct pyopencv_##CLASS_ID##_t \
     { \
         PyObject_HEAD \
         STORAGE v; \
     }; \
-    static PyTypeObject pyopencv_##NAME##_TypeXXX = \
+    static PyTypeObject pyopencv_##CLASS_ID##_TypeXXX = \
     { \
         CVPY_TYPE_HEAD \
-        MODULESTR"."#WNAME, \
-        sizeof(pyopencv_##NAME##_t), \
+        MODULESTR SCOPE"."#NAME, \
+        sizeof(pyopencv_##CLASS_ID##_t), \
     }; \
-    static PyTypeObject * pyopencv_##NAME##_TypePtr = &pyopencv_##NAME##_TypeXXX; \
-    static bool pyopencv_##NAME##_getp(PyObject * self, STORAGE * & dst) \
+    static PyTypeObject * pyopencv_##CLASS_ID##_TypePtr = &pyopencv_##CLASS_ID##_TypeXXX; \
+    static bool pyopencv_##CLASS_ID##_getp(PyObject * self, STORAGE * & dst) \
     { \
-        if (PyObject_TypeCheck(self, pyopencv_##NAME##_TypePtr)) \
+        if (PyObject_TypeCheck(self, pyopencv_##CLASS_ID##_TypePtr)) \
         { \
-            dst = &(((pyopencv_##NAME##_t*)self)->v); \
+            dst = &(((pyopencv_##CLASS_ID##_t*)self)->v); \
             return true; \
         } \
         return false; \
     } \
-    static PyObject * pyopencv_##NAME##_Instance(const STORAGE &r) \
+    static PyObject * pyopencv_##CLASS_ID##_Instance(const STORAGE &r) \
     { \
-        pyopencv_##NAME##_t *m = PyObject_NEW(pyopencv_##NAME##_t, pyopencv_##NAME##_TypePtr); \
+        pyopencv_##CLASS_ID##_t *m = PyObject_NEW(pyopencv_##CLASS_ID##_t, pyopencv_##CLASS_ID##_TypePtr); \
         new (&(m->v)) STORAGE(r); \
         return (PyObject*)m; \
     } \
-    static void pyopencv_##NAME##_dealloc(PyObject* self) \
+    static void pyopencv_##CLASS_ID##_dealloc(PyObject* self) \
     { \
-        ((pyopencv_##NAME##_t*)self)->v.STORAGE::~SNAME(); \
+        ((pyopencv_##CLASS_ID##_t*)self)->v.STORAGE::~SNAME(); \
         PyObject_Del(self); \
     } \
-    static PyObject* pyopencv_##NAME##_repr(PyObject* self) \
+    static PyObject* pyopencv_##CLASS_ID##_repr(PyObject* self) \
     { \
         char str[1000]; \
-        sprintf(str, "<"#WNAME" %p>", self); \
+        sprintf(str, "< " MODULESTR SCOPE"."#NAME" %p>", self); \
```

Comment:
IMHO, patch just switched semantic of macro parameters (switched the first and the second parameter).

Previously:
- `NAME` is internal, C++ code connected identifier.
- `WNAME` is wrapper name.

---

If we really want to do that, then robust patch should rename the macro too to avoid misusing mistakes.

Reply:
Actual name of the `NAME` macro should be a `CLASS_ID`, `WNAME` should be actually `NAME` in this case.